lib/parsers/parseOCLC.py

```diff
@@ -186,7 +197,7 @@ def extractHoldingsLinks(holdings, instance):
 
 
 def _loadURIIdentifier(uri):
-    uriGroup = re.search(r'\/((?:(?!\/).)+)$', uri)
+    uriGroup = re.search(r'\/((?:(?!\/)[^.])+(?=$|\.[a-z]{3,4}$))', uri)
     if uriGroup is not None:
```

Comment:
Q for my understanding: Is this the blacklist block?  So nothing happens if it matches the blacklist, and the format object with source/content_type/links gets added if it doesn't?

Reply:
The blacklist block is actually up at lines 159-162 where it sets a regex for each the providers we want to get ebook files from. This function helps pull useful identifiers from the URIs that we get for the files and passes the result back to the main function. The blacklist is then evaluated in `_matchRegexEbook` and if a match is found an `Item` is added to the instance being parsed.